index.js

```diff
@@ -0,0 +1,112 @@
+const express = require('express')
+const server = express()
+const cors = require('cors')
+const bcrypt = require('bcryptjs')
+const knex = require('knex')
+const knexConfig = require('./knexfile.js')
+const session = require('express-session')
+const KnexSessionStore = require('connect-session-knex')(session)
+
+const db = knex(knexConfig.development)
+
+const sessionConfig = {
+  name: 'monster',
+  secret: 'keep it a secret, keep it safe',
+  cookie: {
+    maxAge: 1000 * 60 * 10, //milliseconds
+    secure: false, // use cookie over https
+    httpOnly: true, // false means can JS access the cookie
+  },
+  resave: false, // resave recreating unchanged sessions
+  saveUnitialized: false, // GDPR compliance
+  store: new KnexSessionStore({
+    knex: db,
+    tablename: 'sessions',
+    sidfieldname: 'sid',
+    clearInterval: 1000 * 60 * 30, // deletes expired sessions
+  })
+}
+
+
+server.use(express.json())
+server.use(cors())
+server.use(session(sessionConfig))
+
+server.get('/', (req, res) => {
+  res.send("Learning Auth")
+})
+
+server.post('/api/register', (req, res) => {
+  let user = req.body
+
+  //hash password
+  const hash = bcrypt.hashSync(user.password, 4)
```

Comment:
good job with hashing (I'm assuming you picked 4 just to help speed up development and that in real life you know to make this number much bigger).